docker/importHistograms.py

```diff
@@ -0,0 +1,136 @@
+BIGQUERY = bigquery.Client()
+FIRESTORE = firestore.Client()
+BATCH_RETRIES = []
+
+
+def commit_batch(batch):
+    try:
+        batch.commit()
+        print(".", end="", flush=True)
+    except DeadlineExceeded:
+        BATCH_RETRIES.append(batch)
+        print("F", end="", flush=True)
+    except:  # noqa
+        print("Unexpected error:", sys.exc_info()[0])
+
+    return FIRESTORE.batch()
+
+
+def import_collection(channel, version):
+
+    query = """
+        SELECT *
+        FROM `{table}`
+        WHERE
+            channel = @channel
+            AND app_version = @app_version
+            AND os IS NOT NULL
+            AND app_build_id IS NOT NULL
+    """.format(
+        table=BQ_TABLE
+    )
+    params = [
+        bigquery.ScalarQueryParameter("channel", "STRING", channel),
+        bigquery.ScalarQueryParameter("app_version", "STRING", version),
+    ]
+    config = bigquery.QueryJobConfig()
+    config.query_parameters = params
+    job = BIGQUERY.query(query, job_config=config)
+
+    print(
+        "Inserting batched documents for {channel}-{version} ...".format(
+            **vars()
+        )
+    )
+
+    i = None
+    batch = FIRESTORE.batch()
+
+    for i, row in enumerate(job):
+
+        if i % 500 == 0 and i > 0:
+            batch = commit_batch(batch)
+
+        doc = {
+            "metric": row.metric,
+            "build_id": row.app_build_id,
+            "os": row.os,
+            "agg_type": row.agg_type,
+            "aggregates": row.aggregates["key_value"],
+        }
+        collection = "{channel}-{version}".format(**vars())
+        # Note: If needed, the equivalent call in nodejs would be the
+        # following, using the "blakejs" lib:
+        # ```
+        # var blake = require("blakejs");
+        # blake.blake2bHex("test")
+        # ```
+        doc_id = hashlib.blake2b(
+            "{metric}-{build_id}-{os}-{agg_type}".format(**doc).encode()
+        ).hexdigest()
+
+        batch.set(FIRESTORE.collection(collection).document(doc_id), doc)
+
+    if i is None:
+        print("No records imported for {channel}-{version}".format(**vars()))
+    else:
+        commit_batch(batch)
+        print(
+            "Done!\nImported {i:,} record(s) for {channel}-{version}".format(
+                **vars()
+            )
+        )
+
+    if BATCH_RETRIES:
+        print("Retrying failed batches...")
+
+        retries = BATCH_RETRIES.copy()
+        BATCH_RETRIES.clear()
+
+        for batch in retries:
+            commit_batch(batch)
+
+        print("Done!")
+
+
+def get_major_version(v):
+    return int(v.split(".")[0])
+
+
+def run():
+    j = json.loads(urllib.request.urlopen(PRODUCT_DETAILS_URL).read())
+    versions = {
+        "release": get_major_version(j["LATEST_FIREFOX_VERSION"]),
+        "beta": get_major_version(j["LATEST_FIREFOX_DEVEL_VERSION"]),
+        "nightly": get_major_version(j["FIREFOX_NIGHTLY"]),
```

Comment:
Just a thought here. In the data, there were a ton of junk channels but I filtered it all out except for the top 6:
"release", "esr", "beta", "aurora", "default", "nightly"

I'm actually not sure what `default` is and whether or not we care about esr or aurora to show up in MDV2. But depending on the answers to those questions, this code here might be affected.